plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/ComparePresenter.java

```diff
@@ -75,45 +80,129 @@ public ComparePresenter(AppContext appContext,
+     *
+     * @param alteredFiles
+     *         ordered touched files
+     * @param currentFile
+     *         file which will be shown first, if null then the first from the list will be shown
+     * @param revisionA
+     *         hash of the first revision or branch.
+     *         If it is set to {@code null}, compare with empty repository state will be performed
+     * @param revisionB
+     *         hash of the second revision or branch.
+     *         If it is set to {@code null}, compare with latest repository state will be performed
+     */
+    public void showCompareBetweenRevisions(final AlteredFiles alteredFiles,
+                                            @Nullable final String currentFile,
+                                            @Nullable final String revisionA,
+                                            @Nullable final String revisionB) {
+        this.alteredFiles = alteredFiles;
+        this.revisionA = revisionA;
+        this.revisionB = revisionB;
+
+        this.compareWithLatest = false;
+
+        findCurrentFile(currentFile);
+        showCompareForCurrentFile();
+    }
+
+    /**
+     * Shows comparison for selected file.
+     * Type of comparison to show depends on {@code compareWithLatest} field.
+     */
+    private void showCompareForCurrentFile() {
+        view.setEnableNextDiffButton(currentFileIndex != (alteredFiles.getFilesQuantity() - 1));
+        view.setEnablePreviousDiffButton(currentFileIndex != 0);
+
+        alteredFiles.getProject()
+                    .getFile(alteredFiles.getFileByIndex(currentFileIndex))
+                    .then(file -> {
+                        if (file.isPresent()) {
+
+                            view.setEnableSaveChangesButton(true);
+
+                            final Container gitDir = getGitDir(file.get());
+                            if (gitDir == null) {
+                                notificationManager.notify(locale.messageFileIsNotUnderGit(file.toString()), WARNING, EMERGE_MODE);
+                                return;
+                            }
+                            final Path relPath = getRelPath(gitDir, file.get());
+
+                            if (compareWithLatest) {
+                                this.comparedFile = file.get();
+
+                                showCompareWithLatestForFile(gitDir.getLocation(),
+                                                             relPath,
+                                                             alteredFiles.getStatusByIndex(currentFileIndex));
+                            } else {
+                                showCompareBetweenRevisionsForFile(gitDir.getLocation(),
+                                                                   relPath,
+                                                                   alteredFiles.getStatusByIndex(currentFileIndex));
+                            }
+
+                        } else { // file is deleted
```

Comment:
To me it is better to keep this comment here, because it is related to whole `else ` branch